draft-ietf-masque-connect-ip.md

```diff
@@ -150,73 +166,141 @@ If a client detects that any of the requirements above are not met by a URI
-"?1" to negotiate support for sending and receiving HTTP capsules
-({{HTTP-DGRAM}}).
+## HTTP/1.1 Request {#req1}
 
-Any 2xx (Successful) response indicates that the proxy is willing to open an IP
-forwarding tunnel between it and the client. Any response other than a
-successful response indicates that the tunnel has not been formed.
+When using HTTP/1.1 {{H1}}, an IP proxying request will meet the following
+requirements:
 
-A proxy MUST NOT send any Transfer-Encoding or Content-Length header fields in a
-2xx (Successful) response to the IP Proxying request. A client MUST treat a
-successful response containing any Content-Length or Transfer-Encoding header
-fields as malformed.
+* the method SHALL be "GET".
 
-The lifetime of the forwarding tunnel is tied to the CONNECT stream. Closing the
-stream (in HTTP/3 via the FIN bit on a QUIC STREAM frame, or a QUIC RESET_STREAM
-frame) closes the associated forwarding tunnel.
+* the request SHALL include a single Host header field containing the origin
+  of the IP proxy.
 
-Along with a successful response, the proxy can send capsules to assign
-addresses and advertise routes to the client ({{capsules}}). The client can also
-assign addresses and advertise routes to the proxy for network-to-network
-routing.
+* the request SHALL include a Connection header field with value "Upgrade"
+  (note that this requirement is case-insensitive as per {{Section 7.6.1 of
+  HTTP}}).
+
+* the request SHALL include an Upgrade header field with value "connect-ip".
+
+An IP proxying request that does not conform to these restrictions is malformed.
+The recipient of such a malformed request MUST respond with an error and SHOULD
+use the 400 (Bad Request) status code.
+
+
+## HTTP/1.1 Response {#resp1}
+
+The IP proxy SHALL indicate a successful response by replying with the
+following requirements:
+
+* the HTTP status code on the response SHALL be 101 (Switching Protocols).
+
+* the response SHALL include a Connection header field with value "Upgrade"
+  (note that this requirement is case-insensitive as per {{Section 7.6.1 of
+  HTTP}}).
+
+* the response SHALL include a single Upgrade header field with value
+  "connect-ip".
+
+* the response SHALL meet the requirements of HTTP responses that start the
+  Capsule Protocol; see {{Section 3.2 of HTTP-DGRAM}}.
+
+If any of these requirements are not met, the client MUST treat this proxying
+attempt as failed and abort the connection.
+
+## HTTP/2 and HTTP/3 Requests {#req23}
+
+When using HTTP/2 {{H2}} or HTTP/3 {{H3}}, IP proxying requests use HTTP
+Extended CONNECT. This requires that servers send an HTTP Setting as specified
+in {{EXT-CONNECT2}} and {{EXT-CONNECT3}} and that requests use HTTP
+pseudo-header fields with the following requirements:
+
+* The :method pseudo-header field SHALL be "CONNECT".
+
+* The :protocol pseudo-header field SHALL be "connect-ip".
+
+* The :authority pseudo-header field SHALL contain the authority of the IP
+  proxy.
```

Comment:
Previously the text said, "contain the host and port". 

a) should we refer [[RFC3986], Section 3.2](https://www.rfc-editor.org/rfc/rfc3986#section-3.2)?
b) port is optional. Is that okay or would there be a reason to require port as we kind of did in the previous text? Just checking to be sure this change is intentional.